ambari-server/src/main/java/org/apache/ambari/server/orm/dao/ClusterDAO.java

```diff
@@ -364,7 +364,7 @@ public ClusterEntity merge(ClusterEntity clusterEntity, boolean flush) {
       entityManager.flush();
     }
 
-    return clusterEntity;
+    return findById(clusterEntity.getClusterId());
```

Comment:
This is odd - are you actually getting back a new instance here?

Reply:
Here, it is not a new instance, it is a just updated instance. clusterEntity is an old instance, after it is saved and flushed, we should return an update one from db, not the old one, it becomes obsolete now.